curriculum/challenges/english/25-front-end-development/lab-magazine-layout/6735a7370e0ae93a4577c689.md

```diff
@@ -0,0 +1,253 @@
+demoType: onClick
+---
+
+# --description--
+
+In this lab, you will design a magazine layout using CSS Grid, including concepts like grid rows and grid columns.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should have a `main` element with the class `magazine-cover`.
+
+2. Your `.magazine-cover` should contain a `header` element for the title with the class `title`.
+
+3. You should have `section` elements for each article and the cover image. Each section should contain a class attribute for each article and the cover image: `feature-article`, `small-article1`, `small-article2`, and `cover-image` respectively.
+
+4. You should set the display property of the `.magazine-cover` to `grid.`
+
+5. The `.magazine-cover` should define a grid layout with the following structure:
+
+   - The title spanning the top row.
+   - Two feature articles and a cover image in the second row.
+   - Two small articles and the cover image in the third row.
+
+6. The `.magazine-cover` should have a `grid-template-columns` property that divides the space into three equal parts.
+
+7. The `.magazine-cover` should have a `grid-template-rows` property. It should be set to `auto` for the first row, and divide equally the space occupied by the remaining two rows.
+
+8. You should add a gap between grid items and set the width of the `.magazine-cover` to occupy most of the viewport width.
+
+**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
+
+# --hints--
+
+You should have a `main` element.
+
+```js
+assert.exists(document.querySelector('main'));
+```
+
+You should have a `.magazine-cover` class for the main element.
+
+```js
+assert.exists(document.querySelector('.magazine-cover'));
+```
+
+You should have a `header` element for the title.
+
+```js
+assert.exists(document.querySelector("main > header"));
+```
+
+You should have a `section` element for each article and the cover image, with classes `feature-article`, `small-article1`, `small-article2`, and `cover-image` respectively.
+
+```js
+const featureArticle = document.querySelector('section.feature-article');
+const smallArticle1 = document.querySelector('section.small-article1');
+const smallArticle2 = document.querySelector('section.small-article2');
+const coverImage = document.querySelector('section.cover-image');
+
+assert.exists(featureArticle);
+assert.exists(smallArticle1);
+assert.exists(smallArticle2);
+assert.exists(coverImage);
+```
+
+The display property of the `.magazine-cover` should be set to `grid`.
+
+```js
+assert.equal(getComputedStyle(document.querySelector('.magazine-cover')).display, 'grid');
+```
+
+Ensure that you have a `grid-template-areas` property with `title` spanning all three columns.
+
+```js
+const magazineCoverStyle = getComputedStyle(document.querySelector('.magazine-cover'));
+assert.include(magazineCoverStyle.gridTemplateAreas, '"title title title"');
```

Comment:
```suggestion
assert.match(magazineCoverStyle.gridTemplateAreas, /^"title title title"/);
```
Otherwise it doesn't check that is the correct row. Same thing for the other rows.